CliFx.Analyzers/ParameterMustBeRequiredExceptLastOneAnalyzer.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Linq;
+using CliFx.Analyzers.ObjectModel;
+using CliFx.Analyzers.Utils.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace CliFx.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class ParameterMustBeRequiredExceptLastOneAnalyzer : AnalyzerBase
+{
+    public ParameterMustBeRequiredExceptLastOneAnalyzer()
```

Comment:
Also rename file and test class
(for consistency with the non-scalar analyzer)

```suggestion
public class ParameterMustBeLastIfNotRequiredAnalyzer : AnalyzerBase
{
    public ParameterMustBeLastIfNotRequiredAnalyzer()
```